Add POD documentation to the example rename code and adds some sanity
checks to the rename operation.

diff -ur perl-5.6.1.orig/eg/rename perl-5.6.1/eg/rename
--- perl-5.6.1.orig/eg/rename	Fri Feb 23 13:57:53 2001
+++ perl-5.6.1/eg/rename	Tue May  8 23:08:57 2001
@@ -1,74 +1,126 @@
-#!/usr/bin/perl
-'di';
-'ig00';
+#!/usr/bin/perl -w
 #
-# $RCSfile: rename,v $$Revision: 4.1 $$Date: 92/08/07 17:20:30 $
+#  This script was developed by Robin Barker (Robin.Barker@npl.co.uk),
+#  from Larry Wall's original script eg/rename from the perl source.
 #
-# $Log:	rename,v $
+#  This script is free software; you can redistribute it and/or modify it
+#  under the same terms as Perl itself.
+#
+# Larry(?)'s RCS header:
+#  RCSfile: rename,v   Revision: 4.1   Date: 92/08/07 17:20:30 
+#
+# $RCSfile: rename,v $$Revision: 1.5 $$Date: 1998/12/18 16:16:31 $
+#
+# $Log: rename,v $
+# Revision 1.5  1998/12/18 16:16:31  rmb1
+# moved to perl/source
+# changed man documentation to POD
+#
+# Revision 1.4  1997/02/27  17:19:26  rmb1
+# corrected usage string
+#
+# Revision 1.3  1997/02/27  16:39:07  rmb1
+# added -v
+#
+# Revision 1.2  1997/02/27  16:15:40  rmb1
+# *** empty log message ***
+#
+# Revision 1.1  1997/02/27  15:48:51  rmb1
+# Initial revision
+#
+
+use strict;
+
+use Getopt::Long;
+$Getopt::Long::ignorecase = 0;
+
+my($verbose, $op, $bad_args);
+
+GetOptions('-v' => \$verbose)			or $bad_args = 1;
+$op = shift					or $bad_args = 1; 
+
+die "Usage: rename [-v] perlexpr [filenames]\n"	if $bad_args;
 
-($op = shift) || die "Usage: rename perlexpr [filenames]\n";
 if (!@ARGV) {
+    print "reading filenames from STDIN\n" if $verbose;
     @ARGV = <STDIN>;
     chop(@ARGV);
 }
+
 for (@ARGV) {
-    $was = $_;
+    my $was = $_;
     eval $op;
     die $@ if $@;
-    rename($was,$_) unless $was eq $_;
+    if( $was eq $_ )	    { }		# ignore quietly
+    elsif( -e $_ )	    { warn  "$was not renamed: $_ already exists\n"; }
+    elsif( rename($was,$_)) { print "$was renamed as $_\n" if $verbose; }
+    else		    { warn  "Can't rename $was $_: $!\n"; }
 }
-##############################################################################
 
-	# These next few lines are legal in both Perl and nroff.
+__END__
+
+=head1 NAME
+
+rename - renames multiple files
+
+=head1 SYNOPSIS
+
+B<rename> S<[ B<-v> ]> I<perlexpr> S<[ I<files> ]>
 
-.00;			# finish .ig
- 
-'di			\" finish diversion--previous line must be blank
-.nr nl 0-1		\" fake up transition to first page again
-.nr % 0			\" start at page 1
-';<<'.ex'; #__END__ ############# From here on it's a standard manual page ############
-.TH RENAME 1 "July 30, 1990"
-.AT 3
-.SH NAME
-rename \- renames multiple files
-.SH SYNOPSIS
-.B rename perlexpr [files]
-.SH DESCRIPTION
-.I Rename
+=head1 DESCRIPTION
+
+C<rename>
 renames the filenames supplied according to the rule specified as the
 first argument.
-The argument is a Perl expression which is expected to modify the $_
+The I<perlexpr> 
+argument is a Perl expression which is expected to modify the C<$_>
 string in Perl for at least some of the filenames specified.
 If a given filename is not modified by the expression, it will not be
 renamed.
 If no filenames are given on the command line, filenames will be read
 via standard input.
-.PP
-For example, to rename all files matching *.bak to strip the extension,
+
+For example, to rename all files matching C<*.bak> to strip the extension,
 you might say
-.nf
 
 	rename 's/\e.bak$//' *.bak
 
-.fi
 To translate uppercase names to lower, you'd use
-.nf
 
 	rename 'y/A-Z/a-z/' *
 
-.fi
-.SH ENVIRONMENT
+=head1 OPTIONS
+
+=over 8
+
+=item B<-v>
+
+Verbose: print names of files successfully renamed.
+
+=back
+
+=head1 ENVIRONMENT
+
 No environment variables are used.
-.SH FILES
-.SH AUTHOR
+
+=head1 AUTHOR
+
 Larry Wall
-.SH "SEE ALSO"
-mv(1)
-.br
-perl(1)
-.SH DIAGNOSTICS
+
+=head1 SEE ALSO
+
+mv(1), perl(1)
+
+=head1 DIAGNOSTICS
+
 If you give an invalid Perl expression you'll get a syntax error.
-.SH BUGS
-.I Rename
-does not check for the existence of target filenames, so use with care.
-.ex
+
+=head1 BUGS
+
+The original
+C<rename>
+did not check for the existence of target filenames,
+so had to be used with care.
+I hope I've fixed that (Robin Barker).
+
+=cut
